main/src/main/scala/sbt/plugins/DependencyTreeKeys.scala

```diff
@@ -35,6 +35,12 @@ trait MiniDependencyTreeKeys {
 
 object MiniDependencyTreeKeys extends MiniDependencyTreeKeys
 
+trait DependencyReportKeys {
+  val dependencyReport = inputKey[Unit]("Generates a report of the project dependencies")
```

Comment:
Not everything is a tree, but here too I'm thinking we should just use `dependencyTree` key in sbt 2.x, and maybe `dependencyTreeNext` or something if we introduce it to 1.x?